### ISSUE TITLE
Failed to remove a trace when retry test in 3.8.1

### ISSUE THREAD

**Issue by QAutomatron** [shazam/fork#154]
Use 3.8.0-SNAPSHOT version. Set flaky strategy to rerun test at least +1, run tests.

If any test failed:
```
Retry requested and allowed. Total retry left: 99 and Single Test case retry left: 0
Failed to delete a file %s
Failed to remove a trace filed for a failed but enqueued again test
```

And in the HTML report, these flaky tests are marked as failed (red).

But there is no such problem in version 3.7.0

**Comment by QAutomatron** [comment]
Still valid in 3.8.1

**Comment by alexstyl** [comment]
Closing, as the issue is fixed in the new 3.9.0
